### ISSUE TITLE
singularity container for brainiak

### ISSUE THREAD

**Issue by peetal** [ReproNim/neurodocker#357]
Hello, 

I'm new to neurodocker and I'm trying to use it to write a recipe for building a singularity container, for the purpose of using Brainiak package on cluster. I know that brainiak is hard to install (it was a hassle to just set it up on my laptop), and I'm encountering some error while building the container with the recipe. And I'm wondering if it is solvable. I would super appreciate any insights, thank you all in advance!

here is the command i used for generating the recipe: 
```
neurodocker generate singularity --base=debian:stretch \
--pkg-manager=apt --install gcc g++ libopenmpi-dev \
--miniconda create_env=neuro \
            conda_install='python=3.7 numpy pandas traits scipy' \
            pip_install='nipype mpi4py psutil brainiak nibabel nilearn matplotlib networkx seaborn' \
--miniconda use_env=neuro conda_install='jupyter' \
--install python3.7-dev

```

While building the container use the recipe, here is the error I got
```
Building wheel for brainiak (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/miniconda-latest/envs/neuro/bin/python3.7 /opt/miniconda-latest/envs/neuro/lib/python3.7/site-packages/pip/_vendor/pep517/_in_process.py build_wheel /tmp/tmp0at4q9vt
       cwd: /tmp/pip-install-wimld_zc/brainiak
  Complete output (250 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/brainiak
  copying brainiak/__init__.py -> build/lib.linux-x86_64-3.7/brainiak
  copying brainiak/isc.py -> build/lib.linux-x86_64-3.7/brainiak
  copying brainiak/image.py -> build/lib.linux-x86_64-3.7/brainiak
  copying brainiak/io.py -> build/lib.linux-x86_64-3.7/brainiak
  creating build/lib.linux-x86_64-3.7/brainiak/reprsimil
  copying brainiak/reprsimil/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/reprsimil
  copying brainiak/reprsimil/brsa.py -> build/lib.linux-x86_64-3.7/brainiak/reprsimil
  creating build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/voxelselector.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/classifier.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/mvpa_voxelselector.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/preprocessing.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  copying brainiak/fcma/util.py -> build/lib.linux-x86_64-3.7/brainiak/fcma
  creating build/lib.linux-x86_64-3.7/brainiak/reconstruct
  copying brainiak/reconstruct/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/reconstruct
  copying brainiak/reconstruct/iem.py -> build/lib.linux-x86_64-3.7/brainiak/reconstruct
  creating build/lib.linux-x86_64-3.7/brainiak/utils
  copying brainiak/utils/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/utils
  copying brainiak/utils/fmrisim.py -> build/lib.linux-x86_64-3.7/brainiak/utils
  copying brainiak/utils/utils.py -> build/lib.linux-x86_64-3.7/brainiak/utils
  creating build/lib.linux-x86_64-3.7/brainiak/eventseg
  copying brainiak/eventseg/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/eventseg
  copying brainiak/eventseg/event.py -> build/lib.linux-x86_64-3.7/brainiak/eventseg
  creating build/lib.linux-x86_64-3.7/brainiak/funcalign
  copying brainiak/funcalign/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/funcalign
  copying brainiak/funcalign/sssrm.py -> build/lib.linux-x86_64-3.7/brainiak/funcalign
  copying brainiak/funcalign/fastsrm.py -> build/lib.linux-x86_64-3.7/brainiak/funcalign
  copying brainiak/funcalign/srm.py -> build/lib.linux-x86_64-3.7/brainiak/funcalign
  copying brainiak/funcalign/rsrm.py -> build/lib.linux-x86_64-3.7/brainiak/funcalign
  creating build/lib.linux-x86_64-3.7/brainiak/factoranalysis
  copying brainiak/factoranalysis/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/factoranalysis
  copying brainiak/factoranalysis/htfa.py -> build/lib.linux-x86_64-3.7/brainiak/factoranalysis
  copying brainiak/factoranalysis/tfa.py -> build/lib.linux-x86_64-3.7/brainiak/factoranalysis
  creating build/lib.linux-x86_64-3.7/brainiak/searchlight
  copying brainiak/searchlight/searchlight.py -> build/lib.linux-x86_64-3.7/brainiak/searchlight
  copying brainiak/searchlight/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/searchlight
  creating build/lib.linux-x86_64-3.7/brainiak/hyperparamopt
  copying brainiak/hyperparamopt/__init__.py -> build/lib.linux-x86_64-3.7/brainiak/hyperparamopt
  copying brainiak/hyperparamopt/hpo.py -> build/lib.linux-x86_64-3.7/brainiak/hyperparamopt
  copying brainiak/utils/grey_matter_mask.npy -> build/lib.linux-x86_64-3.7/brainiak/utils
  running build_ext
  creating tmp
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c /tmp/tmpuh8fl78n.cpp -o tmp/tmpuh8fl78n.o -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c /tmp/tmpcibmfakg.cpp -o tmp/tmpcibmfakg.o -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c /tmp/tmp_eefqft1.cpp -o tmp/tmp_eefqft1.o -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c /tmp/tmpensq11bd.cpp -o tmp/tmpensq11bd.o -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  cythoning brainiak/fcma/cython_blas.pyx to brainiak/fcma/cython_blas.c
  cythoning brainiak/eventseg/_utils.pyx to brainiak/eventseg/_utils.c
  building 'brainiak.factoranalysis.tfa_extension' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/brainiak
  creating build/temp.linux-x86_64-3.7/brainiak/factoranalysis
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c brainiak/factoranalysis/tfa_extension.cpp -o build/temp.linux-x86_64-3.7/brainiak/factoranalysis/tfa_extension.o -g0 -fopenmp -DVERSION_INFO="0.10" -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  In file included from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/pytypes.h:12:0,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/cast.h:13,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/attr.h:13,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/pybind11.h:44,
                   from brainiak/factoranalysis/tfa_extension.cpp:24:
  brainiak/factoranalysis/tfa_extension.cpp: In function ‘PyObject* PyInit_tfa_extension()’:
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:271:20: warning: ‘PyObject* pybind11_init()’ is deprecated: PYBIND11_PLUGIN is deprecated, use PYBIND11_MODULE [-Wdeprecated-declarations]
               return pybind11_init();                                            \
                      ^
  brainiak/factoranalysis/tfa_extension.cpp:242:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(tfa_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:266:22: note: declared here
       static PyObject *pybind11_init();                                          \
                        ^
  brainiak/factoranalysis/tfa_extension.cpp:242:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(tfa_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:271:34: warning: ‘PyObject* pybind11_init()’ is deprecated: PYBIND11_PLUGIN is deprecated, use PYBIND11_MODULE [-Wdeprecated-declarations]
               return pybind11_init();                                            \
                                    ^
  brainiak/factoranalysis/tfa_extension.cpp:242:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(tfa_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:266:22: note: declared here
       static PyObject *pybind11_init();                                          \
                        ^
  brainiak/factoranalysis/tfa_extension.cpp:242:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(tfa_extension) {
   ^~~~~~~~~~~~~~~
  g++ -pthread -shared -B /opt/miniconda-latest/envs/neuro/compiler_compat -L/opt/miniconda-latest/envs/neuro/lib -Wl,-rpath=/opt/miniconda-latest/envs/neuro/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.7/brainiak/factoranalysis/tfa_extension.o -o build/lib.linux-x86_64-3.7/brainiak/factoranalysis/tfa_extension.cpython-37m-x86_64-linux-gnu.so -g0 -fopenmp -DVERSION_INFO="0.10" -std=c++14
  building 'brainiak.fcma.fcma_extension' extension
  creating build/temp.linux-x86_64-3.7/brainiak/fcma
  creating build/temp.linux-x86_64-3.7/brainiak/fcma/src
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c brainiak/fcma/src/fcma_extension.cc -o build/temp.linux-x86_64-3.7/brainiak/fcma/src/fcma_extension.o -g0 -fopenmp -DVERSION_INFO="0.10" -std=c++14
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  brainiak/fcma/src/fcma_extension.cc:60:0: warning: ignoring #pragma simd  [-Wunknown-pragmas]
           #pragma simd
  
  brainiak/fcma/src/fcma_extension.cc: In function ‘void within_subject_norm_native(pybind11::array_t<float, 17>, int)’:
  brainiak/fcma/src/fcma_extension.cc:54:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       for(int v = 0 ; v < nSelectedVoxels*nSubjs ; v++)
                       ~~^~~~~~~~~~~~~~~~~~~~~~~~
  brainiak/fcma/src/fcma_extension.cc:61:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for(int j = 0 ; j < nVoxels ; j++)
                           ~~^~~~~~~~~
  In file included from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/pytypes.h:12:0,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/cast.h:13,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/attr.h:13,
                   from /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/pybind11.h:44,
                   from brainiak/fcma/src/fcma_extension.cc:23:
  brainiak/fcma/src/fcma_extension.cc: In function ‘PyObject* PyInit_fcma_extension()’:
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:271:20: warning: ‘PyObject* pybind11_init()’ is deprecated: PYBIND11_PLUGIN is deprecated, use PYBIND11_MODULE [-Wdeprecated-declarations]
               return pybind11_init();                                            \
                      ^
  brainiak/fcma/src/fcma_extension.cc:87:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(fcma_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:266:22: note: declared here
       static PyObject *pybind11_init();                                          \
                        ^
  brainiak/fcma/src/fcma_extension.cc:87:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(fcma_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:271:34: warning: ‘PyObject* pybind11_init()’ is deprecated: PYBIND11_PLUGIN is deprecated, use PYBIND11_MODULE [-Wdeprecated-declarations]
               return pybind11_init();                                            \
                                    ^
  brainiak/fcma/src/fcma_extension.cc:87:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(fcma_extension) {
   ^~~~~~~~~~~~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include/pybind11/detail/common.h:266:22: note: declared here
       static PyObject *pybind11_init();                                          \
                        ^
  brainiak/fcma/src/fcma_extension.cc:87:1: note: in expansion of macro ‘PYBIND11_PLUGIN’
   PYBIND11_PLUGIN(fcma_extension) {
   ^~~~~~~~~~~~~~~
  g++ -pthread -shared -B /opt/miniconda-latest/envs/neuro/compiler_compat -L/opt/miniconda-latest/envs/neuro/lib -Wl,-rpath=/opt/miniconda-latest/envs/neuro/lib -Wl,--no-as-needed -Wl,--sysroot=/ build/temp.linux-x86_64-3.7/brainiak/fcma/src/fcma_extension.o -o build/lib.linux-x86_64-3.7/brainiak/fcma/fcma_extension.cpython-37m-x86_64-linux-gnu.so -g0 -fopenmp -DVERSION_INFO="0.10" -std=c++14
  building 'brainiak.fcma.cython_blas' extension
  gcc -pthread -B /opt/miniconda-latest/envs/neuro/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/opt/miniconda-latest/envs/neuro/include/python3.7m -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/numpy/core/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -I/tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/pybind11/include -c brainiak/fcma/cython_blas.c -o build/temp.linux-x86_64-3.7/brainiak/fcma/cython_blas.o -g0 -fopenmp -DVERSION_INFO="0.10"
  brainiak/fcma/cython_blas.c:1:2: error: #error Do not use this file, it is the result of a failed Cython compilation.
   #error Do not use this file, it is the result of a failed Cython compilation.
    ^~~~~
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-wimld_zc/brainiak/brainiak/fcma/cython_blas.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
  #  limitations under the License.
  
  # Authors: Yida Wang
  # (Intel Labs), 2016
  
  cimport scipy.linalg.cython_blas as blas
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:18:8: 'scipy/linalg/cython_blas.pxd' not found
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      A = py_a
      cdef float[:, ::1] B
      B = py_b
      cdef float[:, :, ::1] C
      C = py_c
      blas.sgemm(trans_a, trans_b, &M, &N, &K, &alpha, &A[0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:115:8: cimported module has no attribute 'sgemm'
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      beta = py_beta
      cdef float[:, :, ::1] A
      A = py_a
      cdef float[:, ::1] C
      C = py_c
      blas.ssyrk(uplo, trans, &N, &K, &alpha, &A[py_start_voxel, 0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:197:8: cimported module has no attribute 'ssyrk'
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      beta = py_beta
      cdef float[:, ::1] A
      A = py_a
      cdef float[:, :, ::1] C
      C = py_c
      blas.ssyrk(uplo, trans, &N, &K, &alpha, &A[0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:286:8: cimported module has no attribute 'ssyrk'
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      beta = py_beta
      cdef float[:, ::1] A
      A = py_a
      cdef float[:, :, ::1] C
      C = py_c
      blas.sgemm(trans_a, trans_b, &M, &N, &K, &alpha, &A[0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:385:8: cimported module has no attribute 'sgemm'
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      A = py_a
      cdef float[:, ::1] B
      B = py_b
      cdef float[:, :, ::1] C
      C = py_c
      blas.sgemm(trans_a, trans_b, &M, &N, &K, &alpha, &A[0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:477:8: cimported module has no attribute 'sgemm'
  
  Error compiling Cython file:
  ------------------------------------------------------------
  ...
      A = py_a
      cdef float[:, ::1] B
      B = py_b
      cdef float[:, ::1] C
      C = py_c
      blas.sgemm(trans_a, trans_b, &M, &N, &K, &alpha, &A[0, 0], &lda,
         ^
  ------------------------------------------------------------
  
  brainiak/fcma/cython_blas.pyx:559:8: cimported module has no attribute 'sgemm'
  /tmp/pip-build-env-cc4bip_4/normal/lib/python3.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/pip-install-wimld_zc/brainiak/brainiak/eventseg/_utils.pyx
    tree = Parsing.p_module(s, pxd, full_module_name)
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for brainiak
  Building wheel for pydotplus (setup.py) ... done
  Created wheel for pydotplus: filename=pydotplus-2.0.2-py3-none-any.whl size=24566 sha256=f88fbfc81d6ae36b550747eb2bae4bbc46fbae4597feb3dbaa64b83980af3e84
  Stored in directory: /tmp/pip-ephem-wheel-cache-zvx3f8sk/wheels/1e/7b/04/7387cf6cc9e48b4a96e361b0be812f0708b394b821bf8c9c50
  Building wheel for simplejson (setup.py) ... done
  Created wheel for simplejson: filename=simplejson-3.17.0-cp37-cp37m-linux_x86_64.whl size=122414 sha256=521839bd4ca33aa60de095ad763c2132658474e266204e8b534b9fc2eb6f7491
  Stored in directory: /tmp/pip-ephem-wheel-cache-zvx3f8sk/wheels/8a/2d/bb/65cfe3f590a9cbb199f2011a6e4ccecd858844fb04e0dee2e9
  Building wheel for theano (setup.py) ... done
  Created wheel for theano: filename=Theano-1.0.4-py3-none-any.whl size=2667177 sha256=3f5488666709066b5e3da5e0a2568deb21377fe5b0fb4dac585817b41d95f7ff
  Stored in directory: /tmp/pip-ephem-wheel-cache-zvx3f8sk/wheels/33/e0/86/12647586a15bd29c062c9996231380908fb2dcf6a5df1c6f84
  Building wheel for nitime (setup.py) ... done
  Created wheel for nitime: filename=nitime-0.8.1-py3-none-any.whl size=3967490 sha256=caec50b6a27b45bfb5a218b1df0712f80e1a4f915b67d65bb7b8de6cdedf4dad
  Stored in directory: /tmp/pip-ephem-wheel-cache-zvx3f8sk/wheels/fe/e9/5c/3626a4a5dae66799e04734001392429cfc747395278497bb90
  Building wheel for sklearn (setup.py) ... done
  Created wheel for sklearn: filename=sklearn-0.0-py2.py3-none-any.whl size=1315 sha256=398f060ec717f9321d976ee9f47de92f0ed2bc1657aa47617287ecdf42493b1d
  Stored in directory: /tmp/pip-ephem-wheel-cache-zvx3f8sk/wheels/46/ef/c3/157e41f5ee1372d1be90b09f74f82b10e391eaacca8f22d33e
Successfully built mpi4py psutil pydotplus simplejson theano nitime sklearn
Failed to build brainiak
ERROR: Could not build wheels for brainiak which use PEP 517 and cannot be installed directly
FATAL:   failed to execute %post proc: exit status 1
FATAL:   While performing build: while running engine: while running /usr/local/libexec/singularity/bin/starter: exit status 255
```

**Comment by kaczmarj** [comment]
@peetal - thanks for the report. is there a reason you don't want to install brainiak with conda? conda ships pre-compiled binaries, so building from source wouldn't be necessary. let me know if you need to build from source though.

**Comment by peetal** [comment]
Thanks for the reply! I think brainiak has its own channel on conda. Thus when I use `conda_install brainiak` for creating the recipe, I would get the an error message saying `PackageNotFounderror: The following packages are not available from the current channels`, when building the image.

I would like to try to use conda for installing, but I'm not sure how to add a specific channel when generating the recipe. 

Many thanks!

**Comment by kaczmarj** [comment]
You can use the `conda_opts` option in the `--miniconda` step to add `--channel brainiak`. I have outlined the steps below. Because you are only using one conda environment, I would suggest installing into the base environment instead of creating a separate environment.

Let me know if you have any other questions!

```
neurodocker generate singularity \
  --base debian:stretch --pkg-manager apt \
  --miniconda \
      use_env=base \
      conda_install='python=3.7 brainiak jupyter numpy pandas scipy traits' \
      conda_opts='--channel brainiak' > Singularity
# Build image.
sudo singularity build brainiak.sif Singularity
# Test that brainiak can be imported.
singularity run brainiak.sif python -c 'import brainiak'
```

**Comment by peetal** [comment]
@kaczmarj  It worked! Thanks a ton!

**Comment by kaczmarj** [comment]
Excellent! Closing this issue but please open a new one if you run into other problems.

**Comment by SomePersonSomeWhereInTheWorld** [comment]
Why does this error out with `Error: No such option: --base (Possible options: --ants, --base-image, --user)`

```
neurodocker generate singularity   --base-image debian:stretch --pkg-manager apt   --miniconda       use_env=base       conda_install='python=3.8 brainiak jupyter numpy pandas scipy traits'       conda_opts='--channel brainiak' > Singularityneurodocker generate singularity   --base debian:stretch --pkg-manager apt   --miniconda       use_env=base       conda_install='python=3.7 brainiak jupyter numpy pandas scipy traits'       conda_opts='--channel brainiak' > Singularity
Usage: neurodocker generate singularity [OPTIONS]
Try 'neurodocker generate singularity --help' for help.

Error: No such option: --base (Possible options: --ants, --base-image, --user)
```

**Comment by kaczmarj** [comment]
your command  has a second `neurodocker generate` command in it, and that one has a `--base` argument.

**Comment by SomePersonSomeWhereInTheWorld** [comment]
> your command has a second `neurodocker generate` command in it, and that one has a `--base` argument.

Sorry bad copy/paste:

```
neurodocker generate singularity   --base debian:stretch --pkg-manager apt   --miniconda            conda_install='python=3.7 brainiak jupyter numpy pandas scipy traits'       conda_opts='--channel brainiak' > Singularity
Usage: neurodocker generate singularity [OPTIONS]
Try 'neurodocker generate singularity --help' for help.

Error: No such option: --base (Possible options: --ants, --base-image, --user)
```

I'm just using [your suggestion](https://github.com/ReproNim/neurodocker/issues/357#issuecomment-643699681).

**Comment by kaczmarj** [comment]
i think we changed the argument to `--base-name` in 2021 or 2022. my suggest was from 2020 and we went through a refactor since then.

**Comment by SomePersonSomeWhereInTheWorld** [comment]
> singularity build brainiak.sif Singularity

I findally

> i think we changed the argument to `--base-name` in 2021 or 2022. my suggestion was from 2020 and we went through a refactor since then.

I finally got a sample cmd that worked:
` neurodocker generate singularity   --base-image debian  --pkg-manager apt   --miniconda version=latest    conda_install='python=3.7 brainiak jupyter numpy pandas scipy traits'       conda_opts='--channel brainiak' > Singularity`

However there are dependency errors now when trying to use Python 3.7. I tried to use mamba to speed up the installs but that requires a newer version of Python but it appears Brainiak only works on 3.6-3.7, is that still the case? It does not say that on the [Brainiak page](https://brainiak.org/docs/installation.html).

It'd be nice to have a working Singularity container...
